packages/react-codemirror/src/e2e_tests/autoCompletion.spec.tsx

```diff
@@ -234,3 +232,126 @@ test('completes allShortestPaths correctly', async ({ page, mount }) => {
     'MATCH (n) REURN n; MATCH allShortestPaths',
   );
 });
+
+test('shows signature help information on auto-completion for procedures', async ({
+  page,
+  mount,
+}) => {
+  await mount(
+    <CypherEditor
+      schema={testData.mockSchema}
+      featureFlags={{
+        signatureInfoOnAutoCompletions: true,
+      }}
+    />,
+  );
+  const procName = 'apoc.periodic.iterate';
+  const procedure = testData.mockSchema.procedures[procName];
+
+  const textField = page.getByRole('textbox');
+  await textField.fill('CALL apoc.periodic.');
+
+  await expect(page.locator('.cm-tooltip-autocomplete')).toBeVisible();
+
+  const infoTooltip = page.locator('.cm-completionInfo');
+
+  while (!(await infoTooltip.getByText(procName).isVisible())) {
+    await page.keyboard.press('ArrowDown');
+    await page.waitForTimeout(100);
+  }
+
+  await expect(infoTooltip).toContainText(procedure.signature);
+  await expect(infoTooltip).toContainText(procedure.description);
+});
+
+test('shows signature help information on auto-completion for functions', async ({
+  page,
+  mount,
+}) => {
+  await mount(
+    <CypherEditor
+      schema={testData.mockSchema}
+      featureFlags={{
+        signatureInfoOnAutoCompletions: true,
+      }}
+    />,
+  );
+  const fnName = 'apoc.coll.combinations';
+  const fn = testData.mockSchema.functions[fnName];
+
+  const textField = page.getByRole('textbox');
+  await textField.fill('RETURN apoc.coll.');
+
+  await expect(page.locator('.cm-tooltip-autocomplete')).toBeVisible();
+
+  const infoTooltip = page.locator('.cm-completionInfo');
+
+  while (!(await infoTooltip.getByText(fnName).isVisible())) {
+    await page.keyboard.press('ArrowDown');
+    await page.waitForTimeout(100);
```

Comment:
This pattern of a while await with a static timeout strikes me as something that can easily flake. If I understand it correctly what it does is move down the dropdown until the right signature shows? An alternative approach to avoid the waiting for a timeout could be (in pseudo code):

let selectedOption = infoTooltip.getBy(".cm-selected")
while (!selectedOption.textContent.has(fnName)) {
 arrowDown()
 // check that the selection moved
 expect (infoTooltip.getBy(".cm-selected").toNotEqual(selectedOption)
 selected = (infoTooltip.getBy(".cm-selected"))
}

This ensure that's it's not just visible but also selected. It might be fiddly to implement, so if you think it'll work well as is now you can keep it if you prefer